src/test/java/itest/GraphQLIT.java

```diff
@@ -588,6 +602,475 @@ void testNodesHaveIds() throws Exception {
         }
     }
 
+    @Test
+    @Order(9)
+    void testQueryForSpecificTargetsByNames() throws Exception {
+        CompletableFuture<TargetNodesQueryResponse> resp = new CompletableFuture<>();
+
+        String query =
+                String.format(
+                        "query { targetNodes(filter: { names:"
+                            + " [\"service:jmx:rmi:///jndi/rmi://cryostat-itests:9091/jmxrmi\","
+                            + " \"service:jmx:rmi:///jndi/rmi://cryostat-itests:9093/jmxrmi\"] }) {"
+                            + " name nodeType } }");
+        webClient
+                .post("/api/v2.2/graphql")
+                .sendJson(
+                        new JsonObject().put("query", query),
+                        ar -> {
+                            if (assertRequestStatus(ar, resp)) {
+                                resp.complete(
+                                        gson.fromJson(
+                                                ar.result().bodyAsString(),
+                                                TargetNodesQueryResponse.class));
+                            }
+                        });
+        TargetNodesQueryResponse actual = resp.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        List<TargetNode> targetNodes = actual.data.targetNodes;
+
+        int expectedSize = 2;
+
+        assertThat(targetNodes.size(), is(expectedSize));
+
+        TargetNode target1 = new TargetNode();
+        target1.name = "service:jmx:rmi:///jndi/rmi://cryostat-itests:9091/jmxrmi";
+        target1.nodeType = "JVM";
+        TargetNode target2 = new TargetNode();
+        target2.name = "service:jmx:rmi:///jndi/rmi://cryostat-itests:9093/jmxrmi";
+        target2.nodeType = "JVM";
+
+        assertThat(targetNodes, hasItem(target1));
+        assertThat(targetNodes, hasItem(target2));
+    }
+
+    @Test
+    @Order(10)
+    public void testQueryForFilteredActiveRecordingsByNames() throws Exception {
+        // Check preconditions
+        CompletableFuture<JsonArray> listRespFuture1 = new CompletableFuture<>();
+        webClient
+                .get(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, listRespFuture1)) {
+                                listRespFuture1.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
+        JsonArray listResp = listRespFuture1.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        Assertions.assertTrue(listResp.isEmpty());
+        CountDownLatch latch = new CountDownLatch(2);
```

Comment:
I don't think this latch is necessary. Just waiting on the two `CompletableFuture`s in order will do the same thing as what this latch appears to do.